src/ApplePayGateway.php

```diff
@@ -0,0 +1,206 @@
+ *          . PHP_EOL;
+ *        echo 'Response: ' . $authorizeResponse->getResponse() . PHP_EOL;
+ *    } else {
+ *        // Error handling.
+ *        echo 'ERROR: Request not successful.';
+ *        echo 'Status Code: ' . $authorizeResponse->getCode() . PHP_EOL;
+ *        echo 'Status Message: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *    }
+ *
+ *    //Pass authorizeResponse back to the client to validate your merchant and continue with an Apple Pay payment.
+ *    //If successful, the payment sheet should be fully loaded.
+ *    $apple_pay_session = $authorizeResponse->getPaymentSessionObject();
+ *
+ *     // TODO: Add functionality for completeAuthorize() and capture().
+ *     // An opaque Apple Pay Session is returned as a response (expires after 5 mins) and it can be sent to
+ *     // the front end to fully load the payment sheet. This allows the user to optionally configure their
+ *     // payment option and shipping methods (if needed) and submit their payment.
+ * </code>
+ */
+class ApplePayGateway extends AbstractVindiciaGateway
+{
+
+    /**
+     * Get the gateway name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return 'Vindicia ApplePay';
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getPemCertPath()
+    {
+        return $this->getParameter('pemCertPath');
+    }
+
+    /**
+     * A certificate associated with your merchant ID.
+     * Example:
+     * setPemCertPath('./certs/path_to_cert.crt.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setPemCertPath($value)
+    {
+        return $this->setParameter('pemCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPath()
+    {
+        return $this->getParameter('keyCertPath');
+    }
+
+    /**
+     * A key certificate associated with your merchant ID.
+     * Example:
+     * setKeyCertPath('./certs/path_to_cert.key.pem');
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setKeyCertPath($value)
+    {
+        return $this->setParameter('keyCertPath', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getKeyCertPassword()
+    {
+        return $this->getParameter('keyCertPassword');
+    }
+
+    /**
+     * Password associated with your key certificate.
+     * Can be omitted if a password isn't set.
+     * 
+     * @param string $value
+     * @return static
+     */
+    public function setKeyCertPassword($value)
+    {
+        return $this->setParameter('keyCertPassword', $value);
+    }
+
+    /**
+     * Makes request to Apple to set up session between Vimeo and Apple.
+     *
+     * @param array $parameters
+     * @return \Omnipay\Vindicia\Message\ApplePayAuthorizeRequest
+     */
+    public function authorize(array $parameters = array())
```

Comment:
What are all the possible valid parameter values here?  I know you have an example in the class docblock but that would require the user of this function to scroll up and go looking for it. I think it would be better to keep related information close to where it will be used.

If you make all these param values in `ApplePayAuthorizeRequest.php` into consts, you could reference them from here. Just an idea.

Reply:
the normal omnipay documentation style is to document them in the request file (ApplePayAuthorizeRequest.php). it could be useful to add a note here to direct people to that file for the parameters. that'd be helpful to people less familiar with omnipay.